### ISSUE TITLE
ZoomSnap / fractional zoom when using Leaflet with NodeJS

### ISSUE THREAD

**Issue by andylarox** [Leaflet/Leaflet#5770]
Hi, looking for some pointers on this problem I've been having. I'm using Leaflet 1.2.0 but I'm running it on the server with NodeJS (actually using the leaflet-headless wrapper).

It works fine except I can't get fractional zoom to work. Looking at the Leaflet code, there's a line in the getBoundsZoom and _limitZoom  functions that reads:

		    var snap = any3d ? this.options.zoomSnap : 1;

which forces zoomSnap to 1 if the browser doesn't support 3d CSS Transitions. Obviously Node falls into this category so zoomSnap is forced to 1 and we get no fractional zoom.

I've tried bypassing the any3d check, and the fractional zoom then works great for my GeoJSON layer but the tiles layer seems to lose sync wildly. It works with some of my GeoJSON polygons (I can't  figure out why) but generally the tiles layer generally seems to show tiles that are hundreds or thousands of miles due South East.

I'm hoping someone out there is familiar with how it works and can suggest some things to look at? I;ve got to the tearing hair out stage!

**Comment by perliedman** [comment]
I shouldn't say for certain, but I believe the `any3d` check is there since the scaling used with non-integer zoom levels relies on 3D support in the browser.

More specifically, this line is responsible for setting the correct scale transform of a renderer (used for example for GeoJSON vector layers): https://github.com/Leaflet/Leaflet/blob/14c5f1602cdd337df14c37ea8df777578219f744/src/layer/vector/Renderer.js#L97 - as you can see, scaling will only be applied if `any3d` is true. Just bypassing that check in will not make this transform work, so your data will be positioned wrong, since it isn't scaled properly.

In other words: fractional zoom actually requires 3D support to work properly.

**Comment by andylarox** [comment]
Yeah, I think you're correct from what I've seen. Thanks for the reply. I guess I'll have to think of another way!
